### PR TITLE
Prep for scheduling integration

### PR DESCRIPTION
**Pull request by JohnGrisham**
Several changes to get the project ready to be used with nys-scheduling

### CODE REVIEW

File: .npmrc [hunk 1]
```diff
@@ -0,0 +1,2 @@
+@fortawesome:registry=https://npm.fontawesome.com/
+//npm.fontawesome.com/:_authToken=75E4CDCD-60E5-495A-A554-DEE5002A3970
```
Comment (justinneff) [inline .npmrc:2]: This is a public repo and this token should not be committed. It should either be set through an environment variable or globally in the local npm configuration.

Consequently, this token needs to be rotated and replaced in all projects using it as this token is now compromised.

File: dist/components/Calendar/Body/Body.d.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import * as React from 'react';
```
Comment (justinneff) [inline dist/components/Calendar/Body/Body.d.ts]: I would recommend deleting the `dist` folder and committing the deleted files. Then add `dist` back to the `.gitignore`. That would make this PR much easier to review and remove all the generated noise.

File: .npmrc (hunk 1 above)
Comment (danielgary) [reply .npmrc:2]: You're compromised

File: dist/components/Calendar/Body/Body.d.ts (hunk 2 above)
Comment (danielgary) [reply dist/components/Calendar/Body/Body.d.ts]: 100% this

File: src/lib/view.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import { IEventSource } from '../../dist/lib/eventSource'
```
Comment (danielgary) [inline src/lib/view.ts]: Why are we importing from the dist folder?

File: src/components/Calendar/Body/Body.tsx [hunk 4]
```diff
@@ -1,25 +1,47 @@
-import { eachDay, format } from 'date-fns'
 import * as React from 'react'
```
Comment (danielgary) [inline src/components/Calendar/Body/Body.tsx]: I don't know of any reason to do import * as React anywhere.  import React from will work fine and is consistent with other import statements

File: src/components/Calendar/Calendar.tsx [hunk 5]
```diff
@@ -1,31 +1,142 @@
...
+
+  const renderView = React.useCallback((viewComponent: View['component']) => {
+        return viewComponent({ endDate: end, events: props.events, startDate: start, validRange: range })
+      }, [end, start, props.events, range])
+
+  React.useEffect(() => {
+    if(props.weeks && props.weeks > 0 && activeTab === 0) {
+      setNumOfWeeks(props.weeks)
+    }
+  }, [props.weeks])
```
Comment (danielgary) [inline src/components/Calendar/Calendar.tsx]: Need to add a dependency on activeTab here

Comment (danielgary) [reply src/components/Calendar/Calendar.tsx]: ANd setNumOfWEeks

**Review by danielgary — CHANGES_REQUESTED** [review]
From what I can tell, prettier is applied but there are no linting rules.  I added eslint and more specifically the eslint react hooks.   There are several issues with the current use of hooks that need to be fixed.  You can check these by running yarn lint.

**Comment by danielgary** [comment]
I cleaned up this branch to a certain degree, but additional work needs to be done before another review.

File: dist/components/Calendar/Body/Body.d.ts (hunk 2 above)
Comment (danielgary) [reply dist/components/Calendar/Body/Body.d.ts]: I did this

**Review by johnmurphy01 — APPROVED** [review]

**Review by danielgary — APPROVED** [review]

**Comment by danielgary** [comment]
Looks great to me.  I will get this added into ims-react-components

**Pull request merged**